libafl_libfuzzer/libafl_libfuzzer_runtime/src/options.rs

```diff
@@ -0,0 +1,118 @@
+use std::{
+    error::Error,
+    fmt::{Display, Formatter},
+};
+
+use crate::options::RawOption::{Directory, Flag};
+
+enum RawOption<'a> {
+    Directory(&'a str),
+    Flag { name: &'a str, value: i64 },
+}
+
+fn parse_option(arg: &str) -> Option<RawOption> {
+    if arg.starts_with('-') {
+        if let Some((name, value)) = arg.split_at(1).1.split_once('=') {
+            if let Ok(value) = value.parse() {
+                Some(Flag { name, value })
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    } else {
+        Some(Directory(arg))
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum LibfuzzerMode {
+    Fuzz,
+    Merge,
+    Cmin,
+}
+
+#[derive(Debug)]
+pub enum OptionsParseError {
+    MultipleModesSelected,
+}
+
+impl Display for OptionsParseError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            OptionsParseError::MultipleModesSelected => {
+                f.write_str("multiple modes selected in options")
+            }
+        }
+    }
+}
+
+impl Error for OptionsParseError {}
+
+#[derive(Debug)]
+pub struct LibfuzzerOptions<'a> {
+    mode: LibfuzzerMode,
+    dirs: Vec<&'a str>,
+    unknown: Vec<&'a str>,
+}
+
+impl<'a> LibfuzzerOptions<'a> {
```

Comment:
Just out of interest, why not use `Clap` here?

Reply:
libfuzzer has fairly strict rules for how arguments are parsed. To be perfectly plug-and-play (e.g. in CIs and such), we have to do custom option parsing.